docs/source/index.rst

```diff
@@ -34,6 +34,7 @@ Get started using the :ref:`quickstart` or by reading about the :ref:`key concep
     model-registry
     recipes
     gateway/index
+    gateway/databricks
```

Comment:
Don't the databricks-specific mlflow docs usually live on the databricks docs site?

Reply:
Yes, but Corey wants us to publish it on mlflow website as well: https://databricks.atlassian.net/browse/MLOPS-210

Reply:
Can we just create a link to the Databricks AI Gateway docs?

Reply:
> Can we just add a link to the Databricks AI Gateway docs?

Yeah, otherwise we'd have to maintain the databricks AI gateway docs in two places (and two formats).

Reply:
Example:

<img width="1388" alt="image" src="https://github.com/mlflow/mlflow/assets/17039389/40f01e9f-4750-42eb-a448-7cfcfc163ec9">

Reply:
cool, sg!